backend/src/sms/middlewares/sms.middleware.ts

```diff
@@ -153,22 +145,58 @@ export const InitSmsMiddleware = (
     const logMeta = {
       campaignId,
       label,
-      action: 'getCredentialsFromLabel',
+      action: 'getCredentialsFromLabelCampaign',
     }
     try {
       /* Determine if credential name can be used */
       const campaign = campaignId
         ? await SmsService.findCampaign(+campaignId, userId)
         : null
-      const credentialName = campaign?.demoMessageLimit
+      const canUseDemoCredentials = !!campaign?.demoMessageLimit // false if (1) campaign is null or (2) demoMessageLimit is 0
+      if (!canUseDemoCredentials && label === DefaultCredentialName.SMS) {
+        throw new Error(
+          `Campaign cannot use demo credentials. ${label} is not allowed.`
+        )
+      }
+      const credentialName = canUseDemoCredentials
         ? getDemoCredentialName(label)
         : await getCredentialName(label, +userId)
 
       /* Get credential from the name */
-      const credentials = await credentialService.getTwilioCredentials(
+      res.locals.credentials = await credentialService.getTwilioCredentials(
+        credentialName
+      )
+      res.locals.credentialName = credentialName
+      return next()
+    } catch (err) {
+      const errAsError = err as Error
+      logger.error({
+        ...logMeta,
+        message: `${errAsError.stack}`,
+      })
+      return res.status(400).json({ message: `${errAsError.message}` })
+    }
+  }
+
+  const getCredentialsFromLabelTransactional = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<void | Response> => {
+    const { label } = req.body
+    const userId = req.session?.user?.id
+    const logMeta = {
+      label,
+      action: 'getCredentialsFromLabelTransactional',
+    }
+    try {
+      /* Determine if credential name can be used */
+      const credentialName = await getCredentialName(label, +userId)
+
+      /* Get credential from the name */
+      res.locals.credentials = await credentialService.getTwilioCredentials(
```

Comment:
Separating out `getCredentialsFromLabel` into `getCredentialsFromLabelCampaign` and `getCredentialsFromLabelTransactional` so as to avoid weird campaign-error related error if the label is wrong.

The new logic for `getCredentialsFromLabelTransactional` simply checks the credential directly using the `label` and `userId`, so it won't be able to use demo credentials at all. (So the overall logic remains unchanged.)

Current `try/catch` clause is probably too broad but this is consistent with original code, so not going to change in this PR.